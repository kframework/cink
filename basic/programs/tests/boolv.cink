int main() {
  bool b;
  int x;
  cout << "x = "; 
  cin >> x;
  b = x > 5;
  if (b) cout << "true"; else cout << "false";
  cout << endl;
  return 0;
}
