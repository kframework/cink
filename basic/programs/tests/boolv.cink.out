x = false
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        #symScalLoc(1) |-> lambda ( .Decls @ (((((((((bool b) ;) ((int x) 
          ;)) ((cout << "x = ") ;)) ((cin >> x) ;)) ((b = (x > 5)) ;)) (if ( b
           ) ((cout << "true") ;) else ((cout << "false") ;))) ((cout << endl)
           ;)) (return 0 ;)) )
        #symScalLoc(2) |-> false
        #symScalLoc(3) |-> 3
    </store>
</T>
